tests/test_minimal/test_tools/test_backend_and_dataset_configuration/test_helpers/test_configure_backend_overrides.py

```diff
@@ -0,0 +1,77 @@
+"""Unit tests for `get_default_dataset_configurations`."""
+from pathlib import Path
+from typing import Literal
+
+import numcodecs
+import numpy as np
+import pytest
+from hdmf.data_utils import DataChunkIterator
+from hdmf_zarr import NWBZarrIO
+from pynwb import NWBHDF5IO
+from pynwb.testing.mock.base import mock_TimeSeries
+from pynwb.testing.mock.file import mock_NWBFile
+
+from neuroconv.tools.hdmf import SliceableDataChunkIterator
+from neuroconv.tools.nwb_helpers import (
+    configure_backend,
+    get_default_backend_configuration,
+)
+
+
+@pytest.mark.parametrize(
+    "case_name,iterator,iterator_options",
+    [
+        ("unwrapped", lambda x: x, dict()),
+        ("generic", SliceableDataChunkIterator, dict()),
+        ("classic", DataChunkIterator, dict(iter_axis=1, buffer_size=30_000 * 5)),
+        # Need to hardcode buffer size in classic case or else it takes forever...
+    ],
+)
+@pytest.mark.parametrize("backend", ["hdf5", "zarr"])
+def test_simple_time_series_override(
+    tmpdir: Path, case_name: str, iterator: callable, iterator_options: dict, backend: Literal["hdf5", "zarr"]
+):
+    BACKEND_NWB_IO = dict(hdf5=NWBHDF5IO, zarr=NWBZarrIO)
+
+    array = np.zeros(shape=(30_000 * 5, 384), dtype="int16")
+    data = iterator(array, **iterator_options)
+
+    nwbfile = mock_NWBFile()
+    time_series = mock_TimeSeries(name="TestTimeSeries", data=data)
+    nwbfile.add_acquisition(time_series)
+
+    backend_configuration = get_default_backend_configuration(nwbfile=nwbfile, backend=backend)
+    dataset_configuration = backend_configuration.dataset_configurations["acquisition/TestTimeSeries/data"]
+
+    smaller_chunk_shape = (30_000, 64)
+    smaller_buffer_shape = (60_000, 192)
+    higher_gzip_level = 5
+    dataset_configuration.chunk_shape = smaller_chunk_shape
+    dataset_configuration.buffer_shape = smaller_buffer_shape
+
+    higher_gzip_level = 5
+    if backend == "hdf5":
+        dataset_configuration.compression_options = dict(level=higher_gzip_level)
+    elif backend == "zarr":
+        dataset_configuration.compression_options = dict(level=higher_gzip_level)
+
+    configure_backend(nwbfile=nwbfile, backend_configuration=backend_configuration)
+
+    if case_name != "unwrapped":  # TODO: eventually, even this case will be buffered automatically
+        assert nwbfile.acquisition["TestTimeSeries"].data
+
+    nwbfile_path = str(tmpdir / f"test_configure_{backend}_defaults_{case_name}_data.nwb.h5")
```

Comment:
I don't think the string is necessary, right? Also, why the .h5 extension in the path name?

Reply:
It is. `NWB{format}IO` require it to be a `str` or else it complains

The `.h5` is something being debated as a practice; including multiple file suffixes to allow external programs (and us...) to quickly tell what type of backend a given NWB file is without attempting to open it first

For example, now HDFView will always recognize such files without having to go into the settings and adjust it to recognize the `.nwb` suffix

The current opinion seems to be that for the next couple of years, `.nwb` will always be interpreted as being an HDF5 by default, and `.nwb.zarr` will be the identifier of files written using the new Zarr backend. DANDI in particular looks like it will use this suffix approach for deciding what required validations are appropriate (they have extra validations for Zarr assetts)

But if Zarr gains popularity, it might be nice to just get ahead of things and start doing `.nwb.h5` for symmetry along with that to add clarity at a glance

Reply:
import h5py
import os
import random

def create_hdf5_file(filename):
    with h5py.File(filename, 'w') as f:
        dset = f.create_dataset("default", (100,), dtype='i')

def create_random_binary_file(filename):
    with open(filename, 'wb') as f:
        f.write(os.urandom(1024))  # Write 1024 random bytes

def create_nwb_file(filename):
    from pynwb.testing.mock.file import mock_NWBFile
    
    nwbfile = mock_NWBFile()
    # Save the file
    from pynwb import NWBHDF5IO
    with NWBHDF5IO(filename, 'w') as io:
        io.write(nwbfile)
    
    

def is_hdf5_file(filename):
    # Source for magic numbers https://www.loc.gov/preservation/digital/formats/fdd/fdd000229.shtml
    # We should find a better one though
    with open(filename, 'rb') as f:
        file_signature = f.read(8)
    return file_signature == b'\x89HDF\r\n\x1a\n'

# Create files
hdf5_filename = 'test.hdf5'
random_filename = 'random.bin'
nwb_filename = "nwbfile.nwb"

create_nwb_file(nwb_filename)
create_hdf5_file(hdf5_filename)
create_random_binary_file(random_filename)

# Check if files are HDF5
print(f"{hdf5_filename} is HDF5: {is_hdf5_file(hdf5_filename)}")
print(f"{nwb_filename} is HDF5: {is_hdf5_file(nwb_filename)}")
print(f"{random_filename} is HDF5: {is_hdf5_file(random_filename)}")

Reply:
Test this. It works on my system.